geomesa-geojson/geomesa-geojson-api/src/main/scala/org/locationtech/geomesa/geojson/query/GeoJsonQuery.scala

```diff
@@ -171,21 +201,24 @@ object GeoJsonQuery {
   }
 
   /**
-    * Converts a json-path into a filter attribute for geojson simple features. Simple feature type
-    * is assumed to be one defined by @see GeoJsonGtIndex.spec
+    * Read a json geometry object:
+    * { "$geometry" : { "type" : "Point", "coordinates" : [30, 10] } }
     *
-    * @param path json-path
-    * @param dtgPath path of date field in the sft, if any
+    * @param json geometry object
     * @return
     */
-  private def filterAttribute(path: Seq[PathElement], dtgPath: Option[Seq[PathElement]]): PropertyName = {
-    if (path == GeoJsonGeometryPath) {
-      ff.property("geom")
```

Comment:
we need to keep these checks for geometry and date, otherwise we will never use our spatial indices when querying geojson. The geometry and date are extracted during ingestion and used as simple feature attributes, so we have to do the same during query.